core/tests/test_utils.py

```diff
@@ -824,9 +821,10 @@ def put_json(self, url, payload, csrf_token=None, expected_status_int=200):
         self.assertEqual(json_response.status_int, expected_status_int)
         return self._parse_json_response(json_response, expect_errors)
 
-    def get_csrf_token_from_response(self, response):
-        """Retrieve the CSRF token from a GET response."""
-        return re.search(CSRF_REGEX, response.body).group(1)
+    def get_new_csrf_token(self):
+        """Generates CSRF token for test."""
+        response = self.get_json('/csrfhandler')
```

Comment:
@seanlip, I think I've found the problem
getting csrf token after this [statement](https://github.com/oppia/oppia/blob/a572fb9696aeb5c0de192467ee8ae6208ec4cccb/core/tests/test_utils.py#L838) returns
```Response: 302 Moved Temporarily
Cache-Control: no-cache
Content-Type: text/html; charset=utf-8
Location: https://www.google.com/accounts/Logout?continue=http%3A//localhost/csrfhandler
Set-Cookie: ACSID=; expires=Sat, 22 Jun 2019 14:28:40 GMT
Set-Cookie: SACSID=; expires=Sat, 22 Jun 2019 14:28:40 GMT
```

But getting csrf token after the line returns the expected response.
However, we can't get the token before the line because after the login, the token has expired and can't be used for that user.

Reply:
I think this is similar to what happens when we try sending the request in the signup page.

Reply:
Ah ok, yes, this all sounds plausible. So what is the actual sequence of calls that is made when you are poking at pages manually on the dev server, and how does that differ from the sequence of calls that is made in the tests?

Reply:
On the dev server, the csrf token for signup page is sent as part of the data from the signup handler while that of other pages (logged in or not loggedin ) get their tokens from the csrf handler

Reply:
OK, can you mimic that in tests?

Reply:
(I.e. mimic the correct URL endpoint calls)

Reply:
Okay, I'm working on this now

Reply:
@seanlip, from the tests, it seems any call made to the /csrfhandler after signup or login causes the 302 redirect issue.
But getting the token from the signuphandler works

Reply:
Done